src/test/java/com/google/api/codegen/testdata/java_test_test_library.baseline

```diff
@@ -0,0 +1,326 @@
+
+    Assert.assertEquals(request.getShelf(), shelf);
+    Assert.assertEquals(request.getBooks(), books);
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void LibraryServiceTest() {
+    String formattedName = LibraryServiceApi.formatBookName("[SHELF]", "[BOOK]");
+    api.GetBook(formattedName);
+
+    List<GeneratedMessage> requests = serviceHelper.getService().getRequests();
+    Assert.assertEquals(1, requests.size());
+    GetBookRequest request = (GetBookRequest)requests[0];
+
+    Assert.assertEquals(request.getName(), formattedName);
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void LibraryServiceTest() {
+    String formattedName = LibraryServiceApi.formatShelfName("[SHELF]");
+    PageAccessor<Book> response = api.ListBooks(formattedName);
+    for (Book element : response) {
+      // DO NOTHING
+    }
+
+    List<GeneratedMessage> requests = serviceHelper.getService().getRequests();
+    Assert.assertEquals(1, requests.size());
+    ListBooksRequest request = (ListBooksRequest)requests[0];
+
+    Assert.assertEquals(request.getName(), formattedName);
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void LibraryServiceTest() {
+    String formattedName = LibraryServiceApi.formatBookName("[SHELF]", "[BOOK]");
+    api.DeleteBook(formattedName);
+
+    List<GeneratedMessage> requests = serviceHelper.getService().getRequests();
+    Assert.assertEquals(1, requests.size());
+    DeleteBookRequest request = (DeleteBookRequest)requests[0];
+
+    Assert.assertEquals(request.getName(), formattedName);
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void LibraryServiceTest() {
+    String formattedName = LibraryServiceApi.formatBookName("[SHELF]", "[BOOK]");
+    Book book = Book.newBuilder().build();
+    api.UpdateBook(formattedName, book);
+
+    List<GeneratedMessage> requests = serviceHelper.getService().getRequests();
+    Assert.assertEquals(1, requests.size());
+    UpdateBookRequest request = (UpdateBookRequest)requests[0];
+
+    Assert.assertEquals(request.getName(), formattedName);
+    Assert.assertEquals(request.getBook(), book);
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void LibraryServiceTest() {
+    String formattedName = LibraryServiceApi.formatBookName("[SHELF]", "[BOOK]");
+    String formattedOtherShelfName = LibraryServiceApi.formatShelfName("[SHELF]");
+    api.MoveBook(formattedName, formattedOtherShelfName);
+
+    List<GeneratedMessage> requests = serviceHelper.getService().getRequests();
+    Assert.assertEquals(1, requests.size());
+    MoveBookRequest request = (MoveBookRequest)requests[0];
+
+    Assert.assertEquals(request.getName(), formattedName);
+    Assert.assertEquals(request.getOtherShelfName(), formattedOtherShelfName);
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void LibraryServiceTest() {
+
+    PageAccessor<String> response = api.ListStrings();
+    for (String element : response) {
+      // DO NOTHING
+    }
```

Comment:
I think you have to do something with `element` to avoid JVM optimizations. See http://stackoverflow.com/questions/30116664/why-is-this-program-which-loops-many-times-taking-time-when-there-is-a-println